tracer/src/Datadog.Trace/ClrProfiler/AutoInstrumentation/AWS/EventBridge/ContextPropagation.cs

```diff
@@ -0,0 +1,107 @@
+// <copyright file="ContextPropagation.cs" company="Datadog">
+// Unless explicitly stated otherwise all files in this repository are licensed under the Apache 2 License.
+// This product includes software developed at Datadog (https://www.datadoghq.com/). Copyright 2017 Datadog, Inc.
+// </copyright>
+
+#nullable enable
+
+using System;
+using System.Text;
+using Datadog.Trace.DuckTyping;
+using Datadog.Trace.Logging;
+using Datadog.Trace.Propagators;
+
+namespace Datadog.Trace.ClrProfiler.AutoInstrumentation.AWS.EventBridge
+{
+    internal static class ContextPropagation
+    {
+        internal static readonly IDatadogLogger Log = DatadogLogging.GetLoggerFor(typeof(ContextPropagation));
+
+        private const string DatadogKey = "_datadog";
+        private const string StartTimeKey = "x-datadog-start-time";
+        private const string ResourceNameKey = "x-datadog-resource-name";
+        private const int MaxSizeBytes = 256 * 1024; // 256 KB
+
+        // Loops through all entries of the EventBridge event and tries to inject Datadog context into each.
+        public static void InjectTracingContext<TPutEventsRequest>(TPutEventsRequest request, SpanContext context)
+            where TPutEventsRequest : IPutEventsRequest, IDuckType
+        {
+            var entries = request.Entries.Value;
+            if (entries == null)
+            {
+                return;
+            }
+
+            foreach (var entry in entries)
+            {
+                var duckEntry = entry.DuckCast<IPutEventsRequestEntry>();
+                if (duckEntry != null)
+                {
+                    InjectHeadersIntoDetail(duckEntry, context);
+                }
+            }
+        }
+
+        // Tries to add Datadog trace context under the `_datadog` key at the top level of the `detail` field.
+        // `detail` is a string, so we have to manually modify it using a StringBuilder.
+        private static void InjectHeadersIntoDetail(IPutEventsRequestEntry entry, SpanContext context)
+        {
+            var detail = entry.Detail?.Trim() ?? "{}";
+            if (!detail.EndsWith("}"))
+            {
+                // Unable to parse detail string, so just leave it unmodified. Don't inject trace context.
+                Log.Information("Unable to parse detail string. Not injecting trace context.");
```

Comment:
Let's make it a Debug, an info is not really the right choice here.

In general, when we are adding logs for the users, we want to be very cautious in order to not add a new log which they'll be billed for, a debug would be great here.